examples/parser/src/extending_built_in_models_with_json_mypy.py

```diff
@@ -0,0 +1,21 @@
+from pydantic import BaseModel, Json
+
+from aws_lambda_powertools.utilities.parser import event_parser
+from aws_lambda_powertools.utilities.parser.models import APIGatewayProxyEventV2Model
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+class CancelOrder(BaseModel):
+    order_id: int
+    reason: str
+
+
+class CancelOrderModel(APIGatewayProxyEventV2Model):
+    body: Json[CancelOrder]  # type: ignore
```

Comment:
```suggestion
    body: Json[CancelOrder]  # type: ignore[assignment]
```

It's best to ignore this specific assignment differences in typing rather than a wide ignore (all other potential issues like a typo in `bodyyy`)